bin/restrax

```diff
@@ -405,9 +432,131 @@ class ReStrax(daq_core.DataBases):
+        """
+        Here we do difficult software veto stuff. We need to fake that the now recompressed
+        data is the same lineage as the original data. 
+        We need to update the
+        """
+
+        # TODO style of this function is terrible
+        for d in data_docs:
+
+            type_in = d['type']
+            type_orig = d['type'].replace('_sv', '')
+
+            # TODO check if they are more than one
+            d_orig = [d for d in original_docs if d['type'] == type_orig]
+            assert len(d_orig) == 1, " two copies of the same data?! "
+            d_orig = d_orig[0]
+
+            lineage_in = d['meta']['lineage_hash']
+            self.log.debug(f'carlo print d_orig {d_orig}')
+            lineage_orig = d_orig['meta']['lineage_hash']
+
+            dir_in = d['location']
+            dir_orig = d_orig['location']
+            dir_out = self.renamed_path(dir_in)
+
+            backend = strax.FileSytemBackend()
+            md_orig = backend.get_metadata(dir_orig)
+            md_in = backend.get_metadata(dir_in)
+            md_out = backend.get_metadata(dir_out)
+
+            split = dir_out.split('-')
+            split[-2] = split[-2].replace('_sv', '')
+            dir_out_hack = '-'.join(split[:-1]) + '-' + lineage_orig
+            
+            os.rename(dir_out, dir_out_hack)
+            dir_out = dir_out_hack
+
+            # Keep track of plugins used for veto
+            md_out['software_veto_lineage'] = md_out['lineage']
+            
+            # Change filenames in chunk info 
+            for i_chunk in range(len(md_out['chunks'])):
+
+                chunk_name = md_out['chunks'][i_chunk]['filename']
+                data_type, lineage, chunk_number = chunk_name.split('-')
+                data_type = data_type.replace('_sv', '')
+                new_chunk_name = '-'.join([data_type, lineage_orig, chunk_number])
+                md_out['chunks'][i_chunk]['filename'] = new_chunk_name
+                os.rename(os.path.join(dir_out, chunk_name), os.path.join(dir_out, new_chunk_name))
+
+            # Change lineage, kind and type to original raw_records 
+            for key in ['data_kind', 'data_type', 'lineage', 'lineage_hash']:
+                md_out[key] = md_orig[key]
+                
+            # TODO source from FileSaver
+            json_options = dict(sort_keys=True, indent=4) 
+            new_md_json_name = '-'.join([type_orig, lineage_orig, 'metadata.json'])
+            with open(os.path.join(dir_out, new_md_json_name), mode='w') as f:
+                f.write(json.dumps(md_out, **json_options))
+
+            # remove the old metadata file
+            old_md_json_name = '-'.join([type_in, lineage_in, 'metadata.json'])
+            os.remove(os.path.join(dir_out, old_md_json_name))
+            
+            # remove originals from rundb
```

Comment:
probably good if we add this as a function in strax, which just takes an input folder, and some desired output hash?

Reply:
Yep agree, will think about it! Dankje